src/test/java/lambda/part2/exercise/FunctionCombinationExercise.java

```diff
@@ -56,25 +52,25 @@ public void personHasNotEmptyLastNameAndFirstName1() {
     // negate: (T -> boolean) -> (T -> boolean)
     private <T> Predicate<T> negate(Predicate<T> test) {
         // TODO
-        throw new UnsupportedOperationException();
+        return (t) -> !test.test(t);
     }
 
     // TODO
     // and: (T -> boolean, T -> boolean) -> (T -> boolean)
     private <T> Predicate<T> and(Predicate<T> t1, Predicate<T> t2) {
         // TODO
-        throw new UnsupportedOperationException();
+        return (t) -> t1.test(t) && t2.test(t);
     }
 
     @Test
     public void personHasNotEmptyLastNameAndFirstName2() {
         final Predicate<Person> hasEmptyFirstName = p -> p.getFirstName().isEmpty();
         final Predicate<Person> hasEmptyLastName = p -> p.getLastName().isEmpty();
 
-        final Predicate<Person> validateFirstName = null; // TODO use negate
-        final Predicate<Person> validateLastName = null; // TODO use negate
+        final Predicate<Person> validateFirstName = (p) -> negate(hasEmptyFirstName).test(p); // TODO use negate

```

Comment:
Use `negate(hasEmptyFirstName)` instead of `(p) -> negate(hasEmptyFirstName).test(p)`.